tests/testthat/test-dopri.R

```diff
@@ -10,10 +10,17 @@ test_that("can integrate logistic", {
   path <- mode_file("examples/logistic.cpp")
   gen <- mode(path, quiet = TRUE)
   pars <- list(r1 = r[[1]], r2 = r[[2]], K1 = K[[1]], K2 = K[[2]])
-  mod <- gen$new(pars, 0)
+  n_particles <- 5
+  mod <- gen$new(pars, 0, n_particles)
 
-  actual <- vapply(times, function(t) mod$solve(t), numeric(length(y0)))
-  expect_equal(actual, analytic, tolerance = 1e-7)
-  expect_equal(actual, dde, tolerance = 1e-7)
+  actual <- vapply(times, function(t) mod$solve(t), numeric(length(y0) * n_particles))
```

Comment:
A slight change here will make the testing easier:

```
  actual <- vapply(times, function(t) mod$solve(t),
                   matrix(0.0, length(y0), n_particles))
  expect_equal(actual[, 1, ], analytic, tolerance = 1e-7)
  expect_equal(actual[, 1, ], dde, tolerance = 1e-7)
  expect_identical(actual, actual[, rep(1, 5), ])
```

So if each iteration returns a matrix of size `n_state` x `n_particles` then we end up with a 3d matrix `n_state` x `n_particles` x `n_time` which makes poking about in the state a bit nicer later (same pattern in the other tests where you're returning an unstructured vector)